rfd/0097-license-expiry.md

```diff
@@ -0,0 +1,147 @@
+---
+authors: camh (camh@goteleport.com)  
+state: draft
+---
+
+# RFD 97 - License Expiry
+
+## Required approvers
+
+- Engineering: `@r0mant`
+- Product:
+- Security:
+
+## What
+
+Disable SSO enterprise features after license has expired, with a grace period.
+
+## Why
+
+To encourage users to renew their Teleport Enterprise license.
+
+[RFD 84] specified a license expiry warning to encourage users to renew their
+license. This RFD extends that to disable enterprise features if the license
+still has not been renewed after a grace period.
+
+This RFD proposes to disable SAML and OIDC SSO logins only as they are likely
+the most common reason to use Teleport Enterprise. It does not seem likely that
+other enterprise features are being used without also using the enterprise SSO
+connectors, so it is unnecessary work to disable other features.
+
+[RFD 84]: https://github.com/gravitational/teleport/blob/master/rfd/0084-license-expiration-warnings.md
+
+## Success Criteria
+
+The message shown upon license expiry will warn of upcoming disablement of
+enterprise features.
+
+After a 30-day grace period after licence expiry, the license expired banner
+shown on the web UI and in response to `tsh login`, `tsh status` and `tctl` will
+say that enterprise features have been disabled.
+
+After a 30-day grace period after license expiry, login via SAML or OIDC SSO
+will fail with a license expired error message.
+
+## Scope
+
+This RFD applies to self-hosted Teleport Enterprise. Teleport Cloud is currently
+excluded due to missing license expiration.
+
+## Detail
+
+### UX (CLI and web)
+
+The license checker implemented in [RFD 84] already emits alerts at 90 days
+pre-expiry and at expiry. These alerts are shown when a user runs `tsh login`,
+`tsh status` or any `tctl` command, and are displayed in a banner on the web UI.
+
+The alert for license expiry reads:
+
+    Your Teleport Enterprise Edition license has expired on one or more of your
+    auth servers. Please reach out to licenses@goteleport.com to obtain a new
+    license. Inaction may lead to unplanned outage or degraded performance and
+    support.
+
+The last sentence of that message will be changed to:
+
+    Inaction will lead to enterprise features being disabled and limited
```

Comment:
Do we want to be more explicit and give them a date that it will occur ? I can see positives and negatives to that. I think as an Engineer seeing this, I'd want to know how high priority I needed to prompt whichever part of my org was responsible for our Teleport licensing - but at the same time, if they know they have 30 days, they might take longer than if they didn't know.